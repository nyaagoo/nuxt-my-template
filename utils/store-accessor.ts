/* eslint-disable import/no-mutable-exports */

import { Store } from 'vuex'
import { getModule } from 'vuex-module-decorators'
import Counter from '~/store/counter'
import Dog from '~/store/dog'

let counterStore: Counter
let dogStore: Dog

function initializeStores(store: Store<any>): void {
  counterStore = getModule(Counter, store)
  dogStore = getModule(Dog, store)
}

export { initializeStores, counterStore, dogStore }
