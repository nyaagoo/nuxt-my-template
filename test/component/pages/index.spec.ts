import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex, { Store } from 'vuex';
import Index from '~/pages/index.vue';

import { initializeStores, counterStore } from '~/utils/store-accessor';
import Dog from '~/store/dog';
import Counter from '~/store/counter';

let store: Store<any>;
let localVue = createLocalVue();
describe('index', () => {
  beforeAll(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    store = new Vuex.Store({
      modules: { dog: Dog, counter: Counter }
    });
    initializeStores(store);
  });

  test('index.vue is a Vue instance', () => {
    const wrapper = mount(Index, { store, localVue });
    expect(wrapper.isVueInstance).toBeTruthy();
  });

  test('index.vue method get count', () => {
    const wrapper = shallowMount(Index, { store, localVue });
    counterStore.setCounter1(15);
    expect((wrapper.vm as any).count).toBe(15);
  });

  test('index.vue method decrement', () => {
    const wrapper = shallowMount(Index, { store, localVue });
    counterStore.setCounter1(10);
    (wrapper.vm as any).decrement();
    expect(counterStore.counter1).toBe(9);
  });

  test('index.vue method increment', () => {
    const wrapper = shallowMount(Index, { store, localVue });
    counterStore.setCounter1(10);
    (wrapper.vm as any).increment();
    expect(counterStore.counter1).toBe(11);
  });

  test('index.vue method reset', async () => {
    const wrapper = shallowMount(Index, { store, localVue });
    await (wrapper.vm as any).reset();
    expect(counterStore.counter1).toBe(30);
    expect(counterStore.counter2).toBe(30);
  });
});

// TODO: Create vuex Counter module test
