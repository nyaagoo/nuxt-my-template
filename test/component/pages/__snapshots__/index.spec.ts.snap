// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index Snapshot 1`] = `
<div
  class="container"
>
  <div>
    <logo-stub />
    <h1
      class="title"
    >
      nuxt-my-template
    </h1>
    <h2
      class="subtitle"
    >
      hello My groundbreaking Nuxt.js project
    </h2>
    <p>
      
    </p>
    <button>
      +
    </button>
    <button>
      -
    </button>
    <button>
      RESET
    </button>
    <doglist-stub />
    <div
      class="links"
    >
      <a
        class="button--green"
        href="https://nuxtjs.org/"
        target="_blank"
      >
        Documentation
      </a>
      <a
        class="button--grey"
        href="https://github.com/nuxt/nuxt.js"
        target="_blank"
      >
        GitHub
      </a>
    </div>
  </div>
</div>
`;
